src/lib/y2partitioner/widgets/configure.rb

```diff
@@ -275,6 +281,16 @@ def warning_text
         def supported?
           S390_IDS.include?(id) ? Yast::Arch.s390 : true
         end
+
+        # Value for the 'activate' argument of {Reprobe#reprobe}
+        #
+        # For most cases this returns nil, which implies simply honoring the
+        # default behavior.
+        #
+        # @return [Boolean, nil]
+        def activate
+          id == :crypt ? true : nil
```

Comment:
to be honest i start thinking that here it is a bit abusing generic class where it more fits by common class hierarchy. I see there many "child" specific actions and this is one of it ( others are e.g. label defined elsewhere, warning text and others). why not create child class and have factory that create child based on id? If there is reason please document it in Action class description. Because as it is now, reuse single or subset of actions elsewhere is problematic.

Reply:
I don't think there is a reason to avoid a hierarchy other than keeping things simple. These "actions" are not supposed to be reused elsewhere, they are just some very simplistic objects to represent the options of the "Configure" menu-button. They will never be used out of the scope of such button.

Anyway, I will take a look to the possibility of introducing a hierarchy and a factory class.

Reply:
@jreidinger I added a commit which changes this to use a hierarchy of actions. I don't think the factory class helps in this case, so I just reorganized the code a bit to use the new classes directly.

Now is more correct from the OOP point of view, but to be honest I'm not sure if the code has become more readable or simply bigger. :-)